sql/parser/eval.go

```diff
@@ -535,6 +728,15 @@ var cmpOps = map[cmpArgs]func(Datum, Datum) (Datum, error){
 	cmpArgs{EQ, floatType, floatType}: func(left Datum, right Datum) (Datum, error) {
 		return DBool(left.(DFloat) == right.(DFloat)), nil
 	},
+	cmpArgs{EQ, dateType, dateType}: func(left Datum, right Datum) (Datum, error) {
+		return DBool(left.(DDate).Equal(right.(DDate).Time)), nil
+	},
+	cmpArgs{EQ, timestampType, timestampType}: func(left Datum, right Datum) (Datum, error) {
```

Comment:
It looks to me like postgres doesn't store a separate timezone per value.

```
CREATE TABLE t (
  k TIMESTAMP WITH TIME ZONE
);

INSERT INTO t VALUES ('2015-09-01 00:00:00-04'), ('2015-09-01 00:00:00-05');

SELECT * FROM t;
```
This results in:

```
           k
------------------------
 2015-09-01 00:00:00-04
 2015-09-01 01:00:00-04
```

I like the statement: "timezone offset is only used for parsing and presentation". So let's not store the offset and figure out sane rules for what timezone to use for display. I'm fine with adjusting this in a future PR and sprinkling this one with a few TODOs in places where timezones are handled.